src/main/java/com/app/library/listener/BorrowEntityListener.java

```diff
@@ -0,0 +1,35 @@
+package com.app.library.listener;
+
+import com.app.library.model.entity.BorrowEntity;
+import com.app.library.model.enums.BorrowStatus;
+import com.app.library.util.LocalDateTimeWrapper;
+import jakarta.persistence.PostPersist;
+import jakarta.persistence.PrePersist;
+import jakarta.persistence.PreUpdate;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Component
+@AllArgsConstructor
+@Slf4j
+public class BorrowEntityListener {
+
+    private LocalDateTimeWrapper localDateTimeWrapper;
+
+    @PostPersist
+    public void postPersist(BorrowEntity entity) {
+        log.info(String.format("new Borrow record has been saved in database: %s", entity.getId()));
+    }
+
+    @PreUpdate
+    @PrePersist
+    public void setBorrowStatus(BorrowEntity entity) {
+        log.info(String.format("new Borrow record has been saved in database: %s", entity.getId()));
+        if (BorrowStatus.RETURNED.equals(entity.getBorrowStatus())) {
+            entity.setDateOfReturn(localDateTimeWrapper.getLocalDateTimeNow());
```

Comment:
It is better to keep the same invocations in variable :) Please take `localDateTimeWrapper.getLocalDateTimeNow()` into variable in that method

Reply:
Done